sphinx/tutorial/cylc/furthertopics/broadcast.rst

```diff
@@ -0,0 +1,134 @@
+
+Copy the following configuration into a ``suite.rc`` file:
+
+.. code-block:: cylc
+
+   [scheduling]
+       initial cycle point = 1012
+       [[dependencies]]
+           [[[R1]]]
+               graph = wipe_log => announce
+           [[[PT1H]]]
+               graph = announce[-PT1H] => announce
+
+   [runtime]
+       [[wipe_log]]
+           # Delete any files in the suite's "share" directory.
+           script = rm "${CYLC_SUITE_SHARE_DIR}/*"
+
+       [[announce]]
+           script = echo "${CYLC_TASK_CYCLE_POINT} - ${MESSAGE}" >> "${FILE}"
+           [[[environment]]]
+               WORD = ni
+               MESSAGE = We are the knights who say \"${WORD}\"!
+               FILE = "${CYLC_SUITE_SHARE_DIR}/knights"
+
+We now have a suite with an ``announce`` task which runs every hour, writing a
+message to a log file (``share/knights``) when it does so. For the first cycle
+the log entry will look like this::
+
+   10120101T0000Z - We are the knights who say "ni"!
+
+The ``cylc broadcast`` task enables us to change runtime configuration whilst
+the suite is running. For instance we could change the value of the ``WORD``
+environment variable using the command::
+
+   cylc broadcast tutorial-broadcast -n announce -s "[environment]WORD=it"
+
+* ``tutorial-broadcast`` is the name of the suite.
+* ``-n announce`` tells cylc we want to change the runtime configuration of the
+  ``announce`` task.
+* ``-s "[environment]WORD=it"`` changes the value of the ``WORD`` environment
+  variable to ``it``.
+
+Run the suite then try using the ``cylc broadcast`` command to change the
+message::
+
+   cylc run tutorial-broadcast
+   cylc broadcast tutorial-broadcast -n announce -s "[environment]WORD=it"
+
+Inspect the ``share/knights`` file, you should see the message change at
+certain points.
+
+Stop the suite::
+
+   cylc stop tutorial-broadcast
+
+
+In-Situ Example
+---------------
+
+We can call ``cylc broadcast`` from within a task's script. This effectively
+provides the ability for tasks to communicate between themselves.
+
+It is almost always better for tasks to communicate using files but there are
+some niche situations where comminicating via ``cylc broadcast`` is justified.
+This tutorial walks you through using ``cylc broadcast`` to communicate between
+tasks.
+
+.. TODO - examples of this?
+
+Add the following recurrence to the ``dependencies`` section:
+
+.. code-block:: cylc
+
+           [[[PT3H]]]
+               graph = announce[-PT1H] => change_word => announce
+
+The ``change_word`` task runs the ``cylc broadcast`` command to randomly
+change the ``WORD`` environment variable used by the ``announce`` task.
+
+Add the following runtime configuration to the ``runtime`` section:
+
+.. code-block:: cylc
+
+       [[change_word]]
+           script = """
+               # Select random word.
+               IFS=',' read -r -a WORDS <<< $WORDS
+               WORD=${WORDS[$(date +%s) % ${#WORDS[@]}]}
```

Comment:
Is there any way to over-ride or manipulate the syntax highlighting here as the code after the hash-tag symbol on this line (116) is interpreted as a comment and put in the comment colour not the default code colour which looks ugly?

Reply:
I had noticed that, tricky but do-able.

Reply:
Yes, sorry, I imagine you know about a lot if not most or all of the issues I am pointing out but I am just saying what I note in case you don't.